### PR TITLE
Add check for the expected files when install service

### PR DESCRIPTION
**Pull request by anthonydresser**
I also rewrote the install function to use async/await, use rimraf instead of del, and reuse the same function for ssmsmin and sqltoolsservice.

### CODE REVIEW

File: build/gulpfile.sql.js [hunk 1]
```diff
@@ -96,50 +94,39 @@ const formatStagedFiles = () => {
...
+	}
+	await installer.installService(runtime);
+	let stat;
+	for (const file of config.executableFiles) {
+		try {
+			stat = await fs.stat(path.join(serviceInstallFolder, file));
+		} catch (e) { }
+	}
+
+	assert(stat);
```
Comment (Charles-Gagnon) [inline build/gulpfile.sql.js]: If the first one fails but the subsequent ones don't then this won't fail the build. Shouldn't this check be inside the try?

File: build/gulpfile.sql.js [hunk 2]
```diff
@@ -96,50 +94,39 @@ const formatStagedFiles = () => {
...
-	const config = require('../extensions/admin-tool-ext-win/config.json');
-	const runtime = 'Windows_64'; // admin-tool-ext is a windows only extension, and we only ship a 64 bit version, so locking the binaries as such
+async function installService(configPath, runtimId) {
+	const absoluteConfigPath = require.resolve(configPath);
+	const config = require(absoluteConfigPath);
+	const runtime = runtimId || (await platform.getCurrent()).runtimeId;
 	// fix path since it won't be correct
-	config.installDirectory = path.join(__dirname, '..', 'extensions', 'admin-tool-ext-win', config.installDirectory);
+	config.installDirectory = path.join(path.dirname(absoluteConfigPath), config.installDirectory);
+	console.log('install diectory', config.installDirectory);
```
Comment (Charles-Gagnon) [inline build/gulpfile.sql.js]: directory

File: build/gulpfile.sql.js (hunk 1 above)
Comment (anthonydresser) [reply build/gulpfile.sql.js]: We don't care if every file exists, we only care if at least 1 file exists

**Comment by anthonydresser** [comment]
We already have those tests as part of the integration tests.

**Comment by Charles-Gagnon** [comment]
Those will install the service if it doesn't exist though so won't fail. (and we also don't have integration tests for SSMSMin)

**Pull request merged**